markets/perps-market/contracts/storage/PerpsMarket.sol

```diff
@@ -93,33 +94,40 @@ library PerpsMarket {
         }
     }
 
-    function maxLiquidatableAmount(uint128 marketId) internal returns (uint) {
-        Data storage self = load(marketId);
-        uint maxLiquidationValue = maxLiquidationPerSecond(marketId);
+    /**
+     * @dev Returns the max amount of liquidation that can occur based on the market configuration
+     * @notice Based on the configured liquidation window, a trader can only be liquidated for a certain
+     *   amount within that window.  If the amount requested is greater than the amount allowed, the
+     *   smaller amount is returned.  The function also updates its accounting to ensure the results on
+     *   subsequent liquidations work appropriately.
+     */
+    function maxLiquidatableAmount(
+        Data storage self,
+        uint256 requestedLiquidationAmount
+    ) internal returns (uint256 liquidatableAmount) {
+        PerpsMarketConfiguration.Data storage marketConfig = PerpsMarketConfiguration.load(self.id);
+
+        uint maxLiquidationAmountPerSecond = marketConfig.maxLiquidationPerSecond();
         uint timeSinceLastUpdate = block.timestamp - self.lastTimeLiquidationCapacityUpdated;
+        uint maxSecondsInLiquidationWindow = marketConfig.maxSecondsInLiquidationWindow;
 
         self.lastTimeLiquidationCapacityUpdated = block.timestamp.to128();
```

Comment:
You need to save the latest time ONLY if liquidatebleAmount is > 0. Otherwise it will fail in the following scenario (please, add a test case for that scenario):

When there is a series of liquidations every X time, where X is less than `maxSecondsInLiquidationWindow`. 
It will fail to process liquidations after `maxAllowedLiquidationInWindow` is used (due to the else in the if), and will only be able to process liquidations again after a pause of at least `maxSecondsInLiquidationWindow` (when it can enter the first branch of the if)